runtime/integration-tests/src/generic/env.rs

```diff
@@ -247,7 +247,7 @@ pub mod utils {
 			frame_system::CheckNonce::<T>::from(nonce),
 			frame_system::CheckWeight::<T>::new(),
 			pallet_transaction_payment::ChargeTransactionPayment::<T>::from(0),
-			frame_metadata_hash_extension::CheckMetadataHash::<T>::new(true),
+			frame_metadata_hash_extension::CheckMetadataHash::<T>::new(false),
```

Comment:
@lemunozm Unfortunately, multiple of our integration tests fail with `CheckMetadataHash` enabled: https://github.com/centrifuge/centrifuge-chain/actions/runs/9665090255/job/26661355826

I am not fully sure why. Based on the `build.rs`, it is only enabled for `--features=std,metadata-hash`: https://github.com/centrifuge/centrifuge-chain/blob/2e8e44cb814d7fa5c0f18a0a121ff82527909f15/runtime/altair/build.rs#L1-L6

Moreover, it is enabled for `on-chain-release-build`: https://github.com/centrifuge/centrifuge-chain/blob/2e8e44cb814d7fa5c0f18a0a121ff82527909f15/runtime/development/Cargo.toml#L450-L457

I tried both running integration tests with `-F metadata-hash` as well as enabling it for the runtimes in the `integration-tests/Cargo.toml` but ran into polkadot-sdk compiler issues. So it appeared to be the incorrect approach. Upon further investigating the SDK repo, I saw all their integration tests disable the metadata hash. However, this might be due to performance reasons. Apparently, the WASM needs to be built twice if the feature is enabled (see [guide](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/guides/enable_metadata_hash/index.html#enable-metadata-hash-generation)).

Reply:
Yeah I also think it is some missing feature for a polkadot crate. Tried out a lot and still haven't surpassed the error
```
error[E0433]: failed to resolve: could not find `metadata_ir` in `__private`
   --> /Users/william/.cargo/git/checkouts/polkadot-sdk-cff69157b985ed76/c00ca35/substrate/primitives/api/src/lib.rs:797:1
    |
797 | / decl_runtime_apis! {
798 | |     /// The `Core` runtime api that every Substrate runtime needs to implement.
799 | |     #[core_trait]
800 | |     #[api_version(4)]
...   |
827 | |     }
828 | | }
    | |_^ could not find `metadata_ir` in `__private`
```

It can be quickly checked by trying to build any runtime with `on-chain-release` feature. In the Polkadot SDK this works for the template runtime. For us, it doesn't for any runtime.

Reply:
Fails only the test who makes a call to `submit_now()` or `submit_later()`, because those calls sign the transaction. Other calls just mutate the state

Reply:
I'm also stuck in the above error. Quite weird. It seems like `frame-metadata` feature is missing for `sp-api`. But even if you try adding it, the error still happens 🤔. It seems like who add `sp-api` first in the `cargo tree` should add the `frame-metadata` to it.

Reply:
Yeah I have been there as well. Good to know I havent missed anything obvious and sad this doesnt seem to work right now for us. Will probably open an issue in the SDK then.

Reply:
Yeah, Although I think this is more a "rust" problem. It seems unmaintenable that each time you add a feature to a crate, you "need" to propagate that feature to the crates that use the featured gated crate. Ideally, you need to add it even to those crates that don't require that feature. Just for the case that a possible user want to enable it.

Maybe the solution here from polkadot would be to just remove the `frame-metadata` feature. Include `metadata_ir` always. The only reason to add the feature is to avoid the double wasm compilation. But here, it could be avoided.